luigi/__init__.py

```diff
@@ -48,17 +48,32 @@
 from luigi import event
 from luigi.event import Event
 
-from .tools import range  # just makes the tool classes available from command line
-
 
 __all__ = [
     'task', 'Task', 'Config', 'ExternalTask', 'WrapperTask', 'namespace', 'auto_namespace',
     'target', 'Target', 'LocalTarget', 'rpc', 'RemoteScheduler',
     'RPCError', 'parameter', 'Parameter', 'DateParameter', 'MonthParameter',
-    'YearParameter', 'DateHourParameter', 'DateMinuteParameter', 'DateSecondParameter', 'range',
+    'YearParameter', 'DateHourParameter', 'DateMinuteParameter', 'DateSecondParameter',
     'DateIntervalParameter', 'TimeDeltaParameter', 'IntParameter',
     'FloatParameter', 'BoolParameter', 'TaskParameter',
     'ListParameter', 'TupleParameter', 'EnumParameter', 'DictParameter',
     'configuration', 'interface', 'local_target', 'run', 'build', 'event', 'Event',
     'NumericalParameter', 'ChoiceParameter', 'OptionalParameter', 'LuigiStatusCode'
 ]
+
+if not configuration.get_config().has_option('core', 'autoload-range'):
+    import warnings
+    warning_message = '''
+        Autoloading range tasks by default has been deprecated and will be removed in a future version.
+        To get the behavior now add an option to luigi.cfg:
+
+          [core]
+            autoload-range: false
```

Comment:
If this is a new config option, i'd expect to see a docs update.

I have no qualms with this as it doesn't change the default behavior (aside from a warning), but allows powerusers to reduce some noise